Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp

```diff
@@ -480,11 +498,22 @@ SkPaint GraphicsContextSkia::createStrokePaint() const
 void GraphicsContextSkia::setupStrokeSource(SkPaint& paint) const
 {
     if (auto strokePattern = strokeBrush().pattern())
-        paint.setShader(strokePattern->createPlatformPattern({ }, toSkSamplingOptions(imageInterpolationQuality())));
+        setupShader(paint, strokePattern->createPlatformPattern({ }, toSkSamplingOptions(imageInterpolationQuality())));
     else if (auto strokeGradient = strokeBrush().gradient())
-        paint.setShader(strokeGradient->shader(alpha(), strokeBrush().gradientSpaceTransform()));
+        setupShader(paint, strokeGradient->shader(alpha(), strokeBrush().gradientSpaceTransform()));
     else
-        paint.setColor(SkColor(strokeBrush().color().colorWithAlphaMultipliedBy(alpha())));
+        setupColor(paint, SkColor(strokeBrush().color().colorWithAlphaMultipliedBy(alpha())));
+}
+
+void GraphicsContextSkia::setupColor(SkPaint& paint, SkColor color) const
+{
+    paint.setColor4f(SkColor4f::FromColor(color), colorSpace().platformColorSpace().get());
+}
+
+void GraphicsContextSkia::setupShader(SkPaint& paint, sk_sp<SkShader> shader) const
+{
+    if (shader)
+        paint.setShader(shader->makeWithWorkingColorSpace(colorSpace().platformColorSpace()));
```

Comment:
Same here. I think this is a different behavior, in case of null shader we don't set it. I guess it's not a problem because we usually call this for a newly created SkPaint, so it's already null

Reply:
The problem I see here is that every caller such as `paint.setShader(fillPattern->createPlatformPattern({ }, toSkSamplingOptions(imageInterpolationQuality())))` that doesn't have to check if shader is null or not would have to be accompanied by an `if` statement now so that we can safely `->makeWithWorkingColorSpace`.
Alternatively we could make sure that every producer of `sk_sp<SkShader>` but that may lead to crashes if someone adds a new, unchecked producer.

Reply:
I've checked and you're right - `By default, all shaders operate in the destination (surface) color space` - so it looks like in case of shaders it's different than in case of `SkPaint`'s colors. I'll remove the above then.